src/include/misc.h

```diff
@@ -39,24 +39,13 @@
         (v) = (val);        \
         (void)(v);          \
     } while (0);
-#define WT_IGNORE_RET(call)                \
-    do {                                   \
-        uintmax_t __ignored_ret;           \
-        __ignored_ret = (uintmax_t)(call); \
-        WT_UNUSED(__ignored_ret);          \
-    } while (0)
-#define WT_IGNORE_RET_BOOL(call)  \
-    do {                          \
-        bool __ignored_ret;       \
-        __ignored_ret = (call);   \
-        WT_UNUSED(__ignored_ret); \
-    } while (0)
-#define WT_IGNORE_RET_PTR(call)    \
-    do {                           \
-        const void *__ignored_ret; \
-        __ignored_ret = (call);    \
-        WT_UNUSED(__ignored_ret);  \
-    } while (0)
+
+/* 
+ * Explicitly suppress unused return value warning at function call sites. 
+ */
+#define WT_IGNORE_RET(call) ((void)!(call))
```

Comment:
Question: Why do we need to `!` the result before `(void)`ing it?

Reply:
GCC has a bug where `(void)` doesn't cancel the "unused" warning for functions marked as "warn_unused_result" ([link](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509)).

Reply:
Yes and it keeps raising it's head [link](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425#c34).
I had spoken to Andrew out of band, and was going to comment here after I had a chance to validate my reasoning with a compiler expert.